### PR TITLE
Allow even unknown messages, by registration, and allow sending them

### PR DESCRIPTION
**Pull request by rustyrussell**
Closes:  #4963 (ancient)
Closes: #4960

### CODE REVIEW

**Comment by endothermicdev** [comment]
Looks good technically, though it needs a rebase.  I know "It's your funeral," but maybe we should log a single message the first time the node sends a custom even message.  That way there's at least a hint for the operator to grep for?

File: doc/lightning-sendcustommsg.7.md [hunk 1]
```diff
@@ -16,11 +16,9 @@ top, not for direct use by end-users.
 
 The message must be a hex encoded well-formed message, including the 2-byte
 type prefix, but excluding the length prefix which will be added by the RPC
-method. The messages must not use even-numbered types, since these may require
-synchronous handling on the receiving side, and can cause the connection to be
-dropped. The message types may also not use one of the internally handled
+method. The message types may not be one of the internally handled
 types, since that may cause issues with the internal state tracking of
-Core Lightning.
+Core Lightning. We do (as of *v23.11*) allow sending of even types, but note that peers (as per the spec) will disconnect on receiving unknown even types.
```
Comment (endothermicdev) [inline doc/lightning-sendcustommsg.7.md:21]: inconsistent line breaks

**Comment by rustyrussell** [comment]
> Looks good technically, though it needs a rebase. I know "It's your funeral," but maybe we should log a single message the first time the node sends a custom even message. That way there's at least a hint for the operator to grep for?

Good advice!  Made the first one INFO, the DEBUG level after that.

**Comment by rustyrussell** [comment]
Rebased, and added log message.

**Comment by rustyrussell** [comment]
Fixed sanitizer complaint about memcpy NULL, and rebased onto #6802  for flake fixes.

**Pull request merged**